storage/engine/rocksdb.go

```diff
@@ -543,14 +384,14 @@ func (r *rocksDBSnapshot) Defer(func()) {
 
 type rocksDBBatch struct {
 	parent *RocksDB
-	batch  *C.DBBatch
+	batch  *C.DBEngine
```

Comment:
If this was a `C.DBBatch` I wouldn't be able to call the `DB{Get,Put,Delete,etc}` functions. Note that while `DBBatch` is C++, the `DB*` functions are C. I could do some casting, but that feels ugly.